src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -2118,22 +2118,28 @@ public boolean isStringSubclass(DynamicObject string) {
     }
 
     @CoreMethod(names = {"to_sym", "intern"})
-    @ImportStatic({ StringCachingGuards.class, StringOperations.class })
+    @ImportStatic({ StringCachingGuards.class, StringGuards.class, StringOperations.class })
     public abstract static class ToSymNode extends CoreMethodArrayArgumentsNode {
 
-        @Specialization(guards = "equalNode.execute(rope(string), cachedRope)", limit = "getLimit()")
+        @Specialization(guards = { "!isBrokenCodeRange(string)", "equalNode.execute(rope(string), cachedRope)" },
+                limit = "getLimit()")
```

Comment:
Style suggestion: in similar cases, I break the line after `guards = {` so the guards are on their own line(s).
And then the annotation ends on a new line with
`}, limit = "getLimit()")`

Reply:
Fixed.